pinot-common/src/main/java/org/apache/pinot/common/function/FunctionInvoker.java

```diff
@@ -124,7 +140,16 @@ public Class<?> getResultClass() {
    */
   public Object invoke(Object[] arguments) {
     try {
-      return _method.invoke(_instance, arguments);
+      if (_method.isVarArgs()) {
+        Class<?> parameterBaseClass = _parameterClasses[0].getComponentType();
+        // Convert arguments to base class
+        Class<?> klass = Arrays.stream(_method.getParameterTypes()).findFirst().get().getComponentType();
```

Comment:
Support for varargs will already impose overheads, but streaming the parameter types will be unacceptably slow. This operation should be performed once and cached in the `FunctionInvoker`. It's probably better to store a prototype array in a field too, which you can just clone at invocation time.